src/main/java/bussiness_layer/dto/GroupDTO.java

```diff
@@ -2,10 +2,13 @@
 
 import data_layer.domain.Group;
 import lombok.Data;
+import lombok.Getter;
 import lombok.NoArgsConstructor;
+import lombok.Setter;
 
 
-@Data
+@Getter
+@Setter
 @NoArgsConstructor
 public class GroupDTO implements Comparable<GroupDTO>{
```

Comment:
Why do we need to implement Comparable here?

Reply:
i'm using a sorted set in TeachingDTO so i need it

Reply:
You use a sorted set for retrieving students in alphabetically order? if so, we could use jpa repository order by methods.

Reply:
No, i used a sorted set that contains all groups (sem + lab | all groups taking course) that is build using 2 lists from Teaching entity containing groups (sem & lab) that were already retrieved from the database